extensions/kubernetes/vanilla/deployment/src/main/java/io/quarkus/kubernetes/deployment/KubernetesProcessor.java

```diff
@@ -391,6 +399,22 @@ private void applyConfig(Session session, Project project, String target, String
         });
     }
 
+    /**
+     * When there is no registry defined and s2i isn't being used, the only ImagePullPolicy that can work is 'IfNotPresent'.
+     * This case comes up when users want to deploy their application to a cluster like Minikube where no registry is used
+     * and instead they rely on the image being built directly into the docker daemon that the cluster uses.
+     */
+    private ImagePullPolicy determineImagePullPolicy(PlatformConfiguration config,
+            Optional<ContainerImageInfoBuildItem> containerImage, Capabilities capabilities) {
+        ImagePullPolicy imagePullPolicy = config.getImagePullPolicy();
+        if (containerImage.isPresent()
+                && ContainerImageUtil.isRegistryMissingAndNotS2I(capabilities, containerImage.get())
+                && kubernetesDeploy.getAsBoolean()) {
```

Comment:
I don't really get the purpose of calling `kubernetesDeploy.getAsBoolean()`.

Whatever the reason, this needs to change, as this call is performing an actual request to kubernetes. This should never be the case when generating resources. Generated resources should be the same regardless of the if and what is the configured environment on the user side.

Reply:
The idea is that the resource only changes if you are actually deploying, if you are not deploying you get what you configured.

I think it's the best compromise.

Reply:
I would rather decorate what we deploy rather than the physical generated file.

Reply:
That sounds weird to me. `-Dquarkus.kubernetes.deploy=true` should, in my opinion, be the same as `kubectl apply -f target/kubernetes/kubernetes.yml`. Why not do this always?

Reply:
I agree with @Ladicek. If it's not the same thing, I think users will be very confused

Reply:
The idea for me is to surprise users the least. With the proposed PR, users get an indicative warning message and they can inspect what was generated. 
Having what is deployed different from what is generated would cause a lot of surprise I think.

Reply:
I was trying to ask, why not remove the `&& kubernetesDeploy.getAsBoolean()` condition. What is it good for?

Reply:
My idea is to only change the policy when we absolutely have to - which is only when deploying. A user may for some reason want to just generate the resources and in that case I don't see why we should change it.

It's an edge case, so I am happy to go either way

Reply:
So, in addition to "`-Dquarkus.kubernetes.deploy=true` should be the same as `kubectl apply -f target/kubernetes/kubernetes.yml`", I also believe that `target/kubernetes/kubernetes.yml` should be the same regardless of `quarkus.kubernetes.deploy` value. I know, I'm demanding :-)

Reply:
If that isn't possible, then I think I agree with @geoand that it's important to deploy the same thing that is on the filesystem.

Reply:
Well I don't think we can do everything, so we need to find a good compromise. My opinion is obviously that the current PR is the best compromise, but I am open to other ideas.

Reply:
I think that the absence of a registry is enough to trigger the desired behaviour and keeps things simple. Having, the runtime affect in any way the generated resources is going to bite us.

Reply:
So you prefer that I remove the `kubernetesDeploy.getAsBoolean()` check altogether?